zhaquirks/tuya/ts601_door.py

```diff
@@ -0,0 +1,198 @@
+"""Device handler for Tuya Zigbee Vibration Door Sensor.
+
+The device reports:
+- Vibration
+- Open/Closed door/window
+- Remaining battery percentage
+
+Extra details: https://www.aliexpress.com/item/1005004443361928.html?spm=a2g0o.order_list.order_list_main.53.35a81802wiZASs
+"""
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Ota,
+    PowerConfiguration,
+    Scenes,
+    Time,
+)
+from zigpy.zcl.clusters.security import IasZone, ZoneStatus, ZoneType
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+    ZONE_STATE,
+)
+from zhaquirks.tuya import (
+    TUYA_CLUSTER_ID,
+    DPToAttributeMapping,
+    TuyaLocalCluster,
+    TuyaNewManufCluster,
+)
+
+DOOR_HANDLE_DP_ID = 1
+BATTERY_STATE_DP_ID = 3
+VIBRATION_DP_ID = 10
+
+DP_HANDLER_EP_ID = 1
+DOOR_HANDLE_EP_ID = 2
+VIBRATION_EP_ID = 3
+
+
+class CustomBasicCluster(CustomCluster, Basic):
+    """Custom Basic cluster to report power source."""
+
+    _CONSTANT_ATTRIBUTES = {
+        Basic.AttributeDefs.power_source.id: Basic.PowerSource.Battery
+    }
```

Comment:
Not sure we need this, not an issue for now though. Was it shown wrong on the HA device page?
That data comes from the node descriptor though.

Reply:
ZHA displayed the power source as "battery or unknown", which was bothering me a little, and in hopes of fixing that I tried setting it this way. However, it didn't really help, so I guess that is solely based on the device signature.